initializr-generator/src/main/resources/templates/starter-pom.xml

```diff
@@ -12,16 +12,23 @@
 	<description>${description}</description>
 
 	<parent>
+		<% if (useCustomParentPom) { %>
```

Comment:
I think it would be much easier if the parentPom had a default value (the value of what's in the template right now) and have no if at all. You'd just write what's coming from the model. No need for boolean flag, etc.